tests/integration_tests/functional/test_snapshot_advanced.py

```diff
@@ -232,7 +235,9 @@ def create_snapshot_helper(
     # Add disks.
     for scratch in test_drives:
         # Add a scratch 64MB RW non-root block device.
-        scratchdisk = drive_tools.FilesystemFile(tempfile.mktemp(), size=64)
+        scratchdisk = drive_tools.FilesystemFile(
+            tempfile.mktemp(dir=scratch_dir), size=64
```

Comment:
can we use `tmp_path` fixture instead of tempfile? https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html It has some advantages to `tempfile`. 

Alternatively, we could change `TMPDIR` envvar before calling pytest

Reply:
I will look into it, thanks!

Reply:
In fact, you were right! I think the way to go here is putting $TMPDIR pointing to the ramdisk. Then everything is solved and we don't need this hacky solutions for the tests.